zilliqa/src/consensus.rs

```diff
@@ -647,7 +635,20 @@ impl Consensus {
         trace!("checking if block view {} is safe", block.view());
 
         // If the proposed block is safe, vote for it and advance to the next round.
-        if self.check_safe_block(&block, during_sync)? {
+        if self.check_safe_block(&block)? {
+            // If the proposed block is safe but outdated then add to block cache - we may need it later
+            let outdated = block.view() < self.view.get_view();
+            let process_immediately = !outdated || during_sync;
+            if !process_immediately {
+                trace!(
+                    "proposal is outdated: {} < {} but may be useful in the future, buffering",
+                    block.view(),
+                    self.view.get_view()
+                );
+                self.block_store.buffer_proposal(from, proposal)?;
```

Comment:
nit 
```suggestion
                self.block_store.buffer_proposal(from, Proposal::from_parts(block, transactions))?;
```
would avoid the clone at the start.